scraper/src/github-scraper/utils.ts

```diff
@@ -0,0 +1,164 @@
+import path from "path";
+import { octokit } from "./config.js";
+import {
+  Action,
+  ActivityData,
+  ParsedDiscussion,
+  PullRequestEvent,
+} from "./types.js";
+import { mkdir, readFile, writeFile } from "fs/promises";
+
+export const parseISODate = (isoDate: Date) => {
+  return new Date(isoDate);
+};
+
+const userBlacklist = new Set(["dependabot", "snyk-bot", "codecov-commenter"]);
+
+export const isBlacklisted = (login: string): boolean => {
+  return login.includes("[bot]") || userBlacklist.has(login);
+};
+
+export async function calculateTurnaroundTime(event: PullRequestEvent) {
+  const user: string = event.payload.pull_request.user.login;
+  const mergedAt: Date = parseISODate(event.payload.pull_request.merged_at);
+  const createdAt: Date = parseISODate(event.payload.pull_request.created_at);
+
+  const linkedIssues: [string, string][] = [];
+  const linkedIssuesResponse = await octokit.request(
+    `GET ${event.payload?.pull_request?.issue_url}`,
+  );
+  // Fetch url all linked issues url from the response
+  linkedIssues.push([event.repo.name, `#${linkedIssuesResponse.data.number}`]);
```

Comment:
What if the issue was cross-referenced from another repository than the repository of the PR made?